### PR TITLE
build: eliminate Makefile and derive version from git describe

### PR DESCRIPTION
**Pull request by vcaputo**
This eliminates the need for version commits, we just add tags referring
to the merge of the NEWS commit.

The Makefile was just a nuisance and we weren't taking advantage of any
of the dependency/incremental build features.

### CODE REVIEW

File: test [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	src/exec/util
+	src/providers
+	src/providers/cmdline
+	src/providers/ec2
+	src/providers/file
+	src/providers/util
+	src/registry"
+
+echo "Running go fix..."
+fixRes=$(go tool fix -diff $SRC)
```
Comment (crawford) [inline test]: Why save the results?

File: test [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash -eu
+
+source ./build
+
+SRC="	src
```
Comment (crawford) [inline test]: Interesting indentation

File: test [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	src/registry"
+
+echo "Running go fix..."
+fixRes=$(go tool fix -diff $SRC)
+
+echo "Checking gofmt..."
+fmtRes=$(gofmt -d -e -s $SRC)
+if [ -n "${fmtRes}" ]; then
+	echo -e "gofmt checking failed:\n${fmtRes}"
+	exit 255
```
Comment (crawford) [inline test]: 255?

File: test (hunk 2 above)
Comment (vcaputo) [inline test]: Is that a good or bad "interesting"?

File: test (hunk 3 above)
Comment (vcaputo) [inline test]: This script originally came from rkt and I've just been tweaking it since, that's where this came from.

File: test (hunk 2 above)
Comment (crawford) [inline test]: Let's go with bad.

```sh
SRC="
    src
    src/config
    ...
"

**Comment by crawford** [comment]
LGTM

**Pull request merged**